### PR TITLE
[luv] add new port

### PR DESCRIPTION
**Pull request by kylo252**
**Describe the pull request**

add https://github.com/luvit/luv: libuv bindings for lua

- #### What does your PR fix?
  NA

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  All except `uwp`, as it's not supported by `luajit`.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: ports/luv/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+# lunarmodules/lua-compat-5.3 needed as a submodule to configure cmake
+vcpkg_from_github(
+    OUT_SOURCE_PATH LUA_COMPAT53_DIR
+    REPO lunarmodules/lua-compat-5.3
+    REF 8f8e4c6adb43e107f5902e784ef207dc3c8ca06b
+    SHA512 dd8ec5bdc825261c3824ef58eaedbed40e026d30be339b1b9b72530a26882800b21812a92069887d22aa5bb63bf1d687e51b91beca5ae6b9d44a5fe7ff5360d1
+    HEAD_REF master
+)
```
Comment (FrankXie05) [inline ports/luv/portfile.cmake:16]:
```suggestion
```
It should be added as another new port.

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/lua-compat53/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,13 @@
+vcpkg_from_github(
```
Comment (FrankXie05) [inline ports/lua-compat53/portfile.cmake:1]:
```suggestion
#header-only library
vcpkg_from_github(
```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/lua-compat53/portfile.cmake (hunk 2 above)
Comment (FrankXie05) [inline ports/lua-compat53/portfile.cmake:1]:
```suggestion
#header-only library
vcpkg_from_github(
```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Review by FrankXie05 — APPROVED** [review]

File: ports/luv/fix-msvc-build.patch [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+-  set(CompilerFlags
+-    CMAKE_C_FLAGS
+-    CMAKE_C_FLAGS_DEBUG
+-    CMAKE_C_FLAGS_MINSIZEREL
+-    CMAKE_C_FLAGS_RELWITHDEBINFO
+-    CMAKE_C_FLAGS_RELEASE)
+-  foreach(CompilerFlag ${CompilerFlags})
+-    string(REPLACE "/MD" "/MT" ${CompilerFlag} "${${CompilerFlag}}")
+-  endforeach()
+-endif ()
```
Comment (FrankXie05) [inline ports/luv/fix-msvc-build.patch:21]: Can I ask why these flags were deleted? 🤔

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Comment by FrankXie05** [comment]
All features of the ports `lua-compat53`  are tested successfully in the following triplet:

- x86-windows
- x64-windows
- x64-windows-stataic

File: ports/luv/fix-msvc-build.patch (hunk 3 above)
Comment (kylo252) [reply ports/luv/fix-msvc-build.patch:21]: They were causing problems with linking to `MSVCRT`, those flags should be controlled by vcpkg instead, the alternative would've adding [linkage restrictions](https://github.com/microsoft/vcpkg/blob/d9553663ca0c1586043695d92e4d93bdb7440c8b/ports/apsi/portfile.cmake#L1) (as I understand it at least)

**Review by FrankXie05 — APPROVED** [review]

File: ports/lua-compat53/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#header-only library
+vcpkg_from_github(
+		OUT_SOURCE_PATH SOURCE_PATH
+    REPO lunarmodules/lua-compat-5.3
```
Comment (FrankXie05) [inline ports/lua-compat53/portfile.cmake:4]:
```suggestion
    OUT_SOURCE_PATH SOURCE_PATH
    REPO lunarmodules/lua-compat-5.3
```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Review by FrankXie05 — DISMISSED** [review]

File: ports/lua-compat53/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
+  ],
+  "default-features": [
+    "c-api"
+  ],
+  "features": {
+    "c-api": {
```
Comment (dan-shaw) [inline ports/lua-compat53/vcpkg.json:21]: Do we need this feature? It does not seem this feature is used in `portfile.cmake`.

Comment (kylo252) [reply ports/lua-compat53/vcpkg.json:21]: @dan-shaw, sorry for the late reply, the idea is that I'm only providing the C API part of https://github.com/lunarmodules/lua-compat-5.3

> It includes:
> 
> - For writing Lua: The Lua module compat53, which can be require'd from Lua scripts and run in Lua 5.1, 5.2, and 5.3, including a backport of the utf8 module, the 5.3 table module, and the string packing functions straight from the Lua 5.3 sources.
> - For writing C: A C header and file which can be linked to your Lua module written in C, providing some functions from the C API of Lua 5.3 that do not exist in Lua 5.2 or 5.1, making it easier to write C code that compiles with all three versions of liblua.

my thinking is basically:
1. make it clear that this is only providing the C API
2. make it easy to add lua-modules export part

**Pull request merged**